File: amodeus/upstream/models/person.py
from typing import Annotated

from pydantic import Field

from .base import IDModeusModel


class Person(IDModeusModel):
    first_name: Annotated[str, Field(alias="firstName")]
    last_name: Annotated[str, Field(alias="lastName")]
    middle_name: Annotated[str | None, Field(alias="middleName")]
    full_name: Annotated[str, Field(alias="fullName")]

File: src/main/kotlin/me/evgfilim1/amodeus/api/upstream/models/Person.kt
package me.evgfilim1.amodeus.api.upstream.models

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.JsonObject
import me.evgfilim1.amodeus.api.utils.UUID

@Serializable
data class Person(
    override val id: UUID,
    val firstName: String,
    val lastName: String,
    val middleName: String,
    val fullName: String,
    override val _links: JsonObject,
) : IDModeusModel
